### PR TITLE
WT-3056 Allow projected table and join cursors to use primary keys

### PR DESCRIPTION
**Pull request by ddanderson**

### CODE REVIEW

**Comment by ddanderson** [comment]
@michaelcahill could you review please?

File: src/cursor/cur_join.c [hunk 1]
```diff
@@ -1333,7 +1333,7 @@ __wt_curjoin_open(WT_SESSION_IMPL *session,
 	WT_ERR(__wt_scr_alloc(session, 0, &tmp));
 	if (columns != NULL) {
 		WT_ERR(__wt_struct_reformat(session, table,
-		    columns, strlen(columns), NULL, 1, tmp));
+		    columns, strlen(columns), NULL, false, tmp));
```
Comment (agorrod) [inline src/cursor/cur_join.c]: @ddanderson Does this change mean that projections will always include the primary key? I thought the goal was to make it optional?

Comment (ddanderson) [inline src/cursor/cur_join.c]: @agorrod, projections only include columns specified, so keys are not included unless mentioned.  The parameter flag indicates if keys are allowed.  `__wt_struct_reformat` also processes column lists for column groups, which is why that flag is needed.

**Comment by agorrod** [comment]
Thanks for the explanation @ddanderson. lgtm, I'll merge.

**Pull request merged**